### PR TITLE
VM Host Failure - Available Policy Development

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM**
Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccIBMISInstance_withAvailablePolicy'
$ make testacc TESTARGS='-run=TestAccIBMISInstanceTemplate_withAvilablePolicy'

$ make testacc TESTARGS='-run=TestAccIBMISInstanceTemplate_dataBasic'
$ make testacc TESTARGS='-run=TestAccIBMISInstanceTemplates_dataBasic'
$ make testacc TESTARGS='-run=TestAccIBMISInstanceDataSource_basic'
$ make testacc TESTARGS='-run=TestAccIBMISInstancesDataSource_basic'
```
Screenshot 2022-02-27 at 4 24 01 PM
Screenshot 2022-02-27 at 4 28 42 PM
Screenshot 2022-02-27 at 7 46 38 PM
Screenshot 2022-02-27 at 7 58 41 PM
Screenshot 2022-02-27 at 8 00 16 PM
Screenshot 2022-02-27 at 8 08 37 PM

### CODE REVIEW

**Comment by ujjwal-ibm** [comment]
#3631

File: ibm/service/vpc/resource_ibm_is_instance.go [hunk 1]
```diff
@@ -75,13 +75,15 @@ const (
...
-	isInstanceStatusFailed         = "failed"
+	isInstanceStatusRestarting           = "restarting"
+	isInstanceStatusStarting             = "starting"
+	isInstanceActionStatusStopping       = "stopping"
+	isInstanceActionStatusStopped        = "stopped"
+	isInstanceStatusPending              = "pending"
+	isInstanceStatusRunning              = "running"
+	isInstanceStatusFailed               = "failed"
+	isInstanceAvailablePolicyHostFailure = "availability_policy_host_failure"
+	isInstanceHostFailure                = "host_failure"
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_instance.go:86]: where are we using this isInstanceHostFailure

File: ibm/service/vpc/resource_ibm_is_instance.go [hunk 2]
```diff
@@ -158,6 +160,13 @@ func ResourceIBMISInstance() *schema.Resource {
 		),
 
 		Schema: map[string]*schema.Schema{
+			isInstanceAvailablePolicyHostFailure: {
+				Type:        schema.TypeString,
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_instance.go:164]: Will it be both optional and computed..

When user provisions a VM without any availability_policy_host_failure what is the value set in computed

File: ibm/service/vpc/resource_ibm_is_instance_template.go [hunk 3]
```diff
@@ -75,6 +77,14 @@ func ResourceIBMISInstanceTemplate() *schema.Resource {
 		),
 
 		Schema: map[string]*schema.Schema{
+			isInstanceTemplateAvailablePolicyHostFailure: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_instance_template.go:84]: Can you run one scenario create a template using old binary and then use this new binary and run Terraform plan and check if plan is showing any diff (both for instance, template you do this testing)

File: website/docs/r/is_instance.html.markdown [hunk 4]
```diff
@@ -338,7 +339,7 @@ Review the argument references that you can specify for your resource.
 
 ## Attribute reference
 In addition to all argument reference list, you can access the following attribute reference after your resource is created.
-
+- `availability_policy_host_failure` - (String) The availability policy to use for this virtual server instance. The action to perform if the compute host experiences a failure.
```
Comment (hkantare) [inline website/docs/r/is_instance.html.markdown:342]: availability_policy_host_failure is a argument no need to be presen in Attribute Refrence

File: ibm/service/vpc/resource_ibm_is_instance.go (hunk 2 above)
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_instance.go:164]: it is set with teh default value from the API, which is "restart"

File: ibm/service/vpc/resource_ibm_is_instance_template.go (hunk 3 above)
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_instance_template.go:84]: TF plan says No changes. Infrastructure is up-to-date.

File: website/docs/r/is_instance.html.markdown (hunk 4 above)
Comment (deepaksibm) [reply website/docs/r/is_instance.html.markdown]: removed

File: ibm/service/vpc/resource_ibm_is_instance.go (hunk 1 above)
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_instance.go]: removed

**Comment by deepaksibm** [comment]
```
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMISInstance_ResizeBoot'
=== RUN   TestAccIBMISInstance_ResizeBoot
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMISInstance_ResizeBoot (295.74s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     296.864s
```

```
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMISInstance_withAvailablePolicy'
=== RUN   TestAccIBMISInstance_withAvailablePolicy
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
no message sent
--- PASS: TestAccIBMISInstance_withAvailablePolicy (474.49s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     476.042s
```

**Comment by deepaksibm** [comment]
Hi @dprosper , i checked with the latest release 1.40.0-beta0 and i could see `availability_policy_host_failure` being accepted and updated in the state file. Could you please verify once if you're using the same latest version ?

image

**Comment by dprosper** [comment]
Hello @deepaksibm yes I do see it and it is working in the beta, the question is **production release and docs**.  should docs be updated when it is not available in the production release? 

There is no indication in the docs that I can tell that this is a beta feature
installing the beta release is not by default, I had to choose.

I think the docs should reflect that it is a new flag accepted in the beta only  
or
there should be separate beta docs
or 
....

**Comment by astha-jain** [comment]
Hi @dprosper VM Host is GA. and this release 1.40.0-beta0 is not for supporting Beta features but it is a **pre-release** version for GA features going to be available in this month's Terraform provider release.

**Pull request merged**